acceptance-tests/tests/src/test/java/org/hyperledger/besu/tests/web3j/privacy/OnChainPrivacyAcceptanceTest.java

```diff
@@ -103,6 +106,96 @@ public void canAddParticipantToGroup() {
     charlie.verify(privateTransactionVerifier.existingPrivateTransactionReceipt(commitmentHash));
   }
 
+  @Test
+  public void aliceCanAddRemoveAddBob() {
+    final String privacyGroupId = createOnChainPrivacyGroup(alice);
+
+    checkOnChainPrivacyGroupExists(privacyGroupId, alice);
+
+    alice.execute(privacyTransactions.privxLockPrivacyGroupAndCheck(privacyGroupId, alice));
+    alice.execute(privacyTransactions.addToPrivacyGroup(privacyGroupId, alice, bob));
+
+    checkOnChainPrivacyGroupExists(privacyGroupId, alice, bob);
+
+    final EventEmitter eventEmitter = deployEventEmitterContract(privacyGroupId, alice);
+
+    final PrivateTransactionReceipt receiptWhileBobMember =
+        setEventEmitterValue(privacyGroupId, eventEmitter, VALUE_WHILE_BOB_MEMBER);
+
+    checkEmitterValue(alice, privacyGroupId, eventEmitter, VALUE_WHILE_BOB_MEMBER);
+    checkEmitterValue(bob, privacyGroupId, eventEmitter, VALUE_WHILE_BOB_MEMBER);
+
+    alice.execute(privacyTransactions.removeFromPrivacyGroup(privacyGroupId, alice, bob));
+
+    checkOnChainPrivacyGroupExists(privacyGroupId, alice);
+
+    final PrivateTransactionReceipt receiptWhileBobRemoved =
+        setEventEmitterValue(privacyGroupId, eventEmitter, VALUE_WHILE_BOB_REMOVED);
+    checkEmitterValue(alice, privacyGroupId, eventEmitter, VALUE_WHILE_BOB_REMOVED);
+    checkEmitterValue(
+        bob,
+        privacyGroupId,
+        eventEmitter,
+        VALUE_WHILE_BOB_MEMBER); // bob did not get the last transaction
```

Comment:
This seems to be repeated pattern in this test.
1. Change the contract value
2. Check the expected value for members.

Any chance that we can encapsulate this logic into a method and reuse across this test? It should make the test more readable.

Something like:
```
changeValueAndCheck(privacyGroupId, contract, members, expectedValue);
```

Reply:
Done